packages/mpg123.rb

```diff
@@ -1,41 +1,28 @@
-require 'package'
+require 'buildsystems/autotools'
 
-class Mpg123 < Package
+class Mpg123 < Autotools
   description 'Fast console MPEG Audio Player and decoder library.'
   homepage 'http://www.mpg123.org/'
-  version '1.29.3'
-  compatibility 'all'
+  version '1.32.3'
   license 'GPL-2 and LGPL-2.1'
-  source_url 'https://sourceforge.net/projects/mpg123/files/mpg123/1.29.3/mpg123-1.29.3.tar.bz2'
-  source_sha256 '963885d8cc77262f28b77187c7d189e32195e64244de2530b798ddf32183e847'
+  compatibility 'x86_64 aarch64 armv7l'
```

Comment:
I had issues building `mpg123` in your `i686` docker image, but there are `i686` [packages](https://centos.pkgs.org/7/centos-x86_64/mpg123-devel-1.25.6-1.el7.i686.rpm.html) for `mpg123`, so maybe its a docker issue? Could be something with the emulation or cross-compilation or something. 

Did you compile this on real hardware? If not, could you try it?

Reply:
As a last resort, compilation on the docker image worked when adding `--with-cpu=i386`, but this looks like a symptom of a root issue with the docker image, as it appears to be presenting itself as `x86_64` in some way.

Reply:
The `i686` docker container reports an arch of `x86_64` from both the `arch` command and `uname -m`. Manually overriding the configure script to use a target of `i686-cros-linux-gnu`, which is what `crew` would use on an `i686` arch, works. 

Given that `--with-cpu=i386` disables some optimisations, I think it'd make sense to revisit this in future when we find out the issue with the docker image.